### PR TITLE
[MI-2669]: Created hook to fetch and load filters

### PR DESCRIPTION
**Pull request by avas27JTG**
Created hook to fetch and load filters

### CODE REVIEW

File: webapp/src/hooks/useLoadFilters.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import {useCallback, useEffect} from 'react';
+
+import {filterLabelValuePairAll} from 'pluginConstants/common';
+import {formLabelValuePairs} from 'utils';
+
+import pluginConstants from 'pluginConstants';
+
+import usePluginApi from './usePluginApi';
+
+type UseLoadFiltersParams = {
```
Comment (ayusht2810) [inline webapp/src/hooks/useLoadFilters.ts:10]:
```suggestion
type UseLoadFiltersProps = {
```

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/containers/modals/SubscribeModal/filters/pipelines.tsx [hunk 2]
```diff
@@ -286,8 +223,8 @@ const PipelinesFilter = ({
                             <Dropdown
                                 placeholder='Stage'
                                 value={selectedRunStageId}
-                                onChange={handleSelectRunStageId}
-                                options={getRunStageIdOptions()}
+                                onChange={(newValue) => handleSetFilter('runStageId', newValue)}
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/modals/SubscribeModal/filters/pipelines.tsx:226]: do we have constants for these keys?

File: webapp/src/hooks/useLoadFilters.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            ([{...filterLabelValuePairAll}, ...formLabelValuePairs('displayValue', 'value', filterList, ['[Any]'])]) :
+            [pluginConstants.common.filterLabelValuePairAll]
+    ), [isLoading, isSuccess]);
+
+    useEffect(() => {
+        if (isError && !isSuccess) {
+            setIsFiltersError(true);
+        } else {
+            setIsFiltersError(false);
+        }
```
Comment (raghavaggarwal2308) [inline webapp/src/hooks/useLoadFilters.ts:47]:
```suggestion
        if (isError && !isSuccess) {
            setIsFiltersError(true);
            return;
        } 
        
        setIsFiltersError(false);
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/hooks/useLoadFilters.ts (hunk 1 above)
Comment (avas27JTG) [reply webapp/src/hooks/useLoadFilters.ts:10]: this is a react hook, we name props only for react components

File: webapp/src/containers/modals/SubscribeModal/filters/pipelines.tsx (hunk 2 above)
Comment (avas27JTG) [reply webapp/src/containers/modals/SubscribeModal/filters/pipelines.tsx:226]: no we use type string literals

File: webapp/src/hooks/useLoadFilters.ts (hunk 3 above)
Comment (avas27JTG) [reply webapp/src/hooks/useLoadFilters.ts:47]: we should not use `return` this way inside `useEffect` as it used mainly for cleanup function

https://stackoverflow.com/questions/56800694/what-is-the-expected-return-of-useeffect-used-for

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**